internal/storage/v2/elasticsearch/tracestore/from_dbmodel.go

```diff
@@ -134,12 +134,17 @@ func dbTagsToAttributes(tags []dbmodel.KeyValue, attributes pcommon.Map) {
 	for _, tag := range tags {
 		tagValue, ok := tag.Value.(string)
 		if !ok {
-			// We have to do this as we are unsure that whether bool will be a string or a bool
-			tagBoolVal, boolOk := tag.Value.(bool)
-			if boolOk && tag.Type == dbmodel.BoolType {
-				attributes.PutBool(tag.Key, tagBoolVal)
+			// We need to handle the numbers in a different way as JSON takes all numbers as float64
```

Comment:
I don't understand this branching. We have an enum indicating the type, why can't we do a switch statement and handing differences in there, instead of splitting that logic with if statements?